### DISCUSSION TITLE
Add stable tag/reduce build frequency

### DISCUSSION THREAD

**Discussion by FuzzyMistborn** [hertzg/rtl_433_docker#30] [Q&A] [answered]
Would it be possible to reduce the build frequency from daily to weekly? Or add a stable tag that gets updated less frequently? I use DIUN to notify me of container updates and it gets triggered every night which gets annoying.  What I've done on other repos is create a separate build file but before I submitted the PR wanted to make sure that was ok.

**Comment by hertzg** [comment]
The images are being built daily, but unless Dockerfile has been modified it re-uses the cached version and resulting image should always be the same.

The only exception should be the `master` builds, they take the `master` branch from merbanan/rtl_433 and build it nightly.


Which tag are you checking updates for?

**Comment by FuzzyMistborn** [comment]
`hertzg/rtl_433:latest` is what I run.

**Comment by hertzg** [comment]
Latest is an alias for latest release of rtl433 based on alpine image. It should change when a new version of rtl433 is released.

Maybe you can configure DUIN. To check for image hash instead of last push date? (Not sure if that's the case)

**Comment by hertzg** [comment]
I'll be closing this for now as this is not an issue more of a discussion. Feel free to open a discussion with this topic if you have suggestion.

**Comment by hertzg** [comment]
I've converted this to a discussion if you have suggestions feel free to comment here.

**Comment by FuzzyMistborn** [comment]
Oh that's cool that you can do that, thanks!

I looked into configuring DIUN but no luck.  From what I understand it goes off the hash so if that changes the notification goes off.  Maybe I could use a different tag from `latest` but just looking through the tag history on DockerHub seems like all/most get updated regularly.

**Reply by hertzg** [reply]
If I were to speculate this could be some caching issue (not sure), The hashes for rtl_433 releases (git tags) should only change if the base image or any of the installed dependencies gets updated.

Release tags could get updated because:
* Base image gets updated (alpine, debian and their versions respectively)
* Any of the installed dependencies get updated (using apk and apt respectively)

So far I've noticed that the some builds are instant because they are fully cached and don't change but some of them still get re-built, not exactly sure why but right now I'm trying to fix the [`gha` cache exporter issue](https://github.com/hertzg/rtl_433_docker/pull/43)

**Comment by chennin** [comment]
Hi @hertzg  ! I noticed I'm no longer getting daily updates for `hertzg/rtl_433:latest`, so checking and updating this discussion: I think https://github.com/hertzg/rtl_433_docker/pull/129 has improved caching a lot?

**Comment by hertzg** [comment, marked as answer]
Looked into the scheduled builds after PR #129 (the caching fix) was merged on 2026-01-11:

| Date | Run ID | Duration | Cached Layers | rtl_433 SHA |
|------|--------|----------|---------------|-------------|
| 2026-01-12 | [20904065179](https://github.com/hertzg/rtl_433_docker/actions/runs/20904065179) | - | - | (cancelled) |
| 2026-01-13 | [20939337629](https://github.com/hertzg/rtl_433_docker/actions/runs/20939337629) | 2m 14s | 13/13 | `8d92cdd` |
| 2026-01-14 | [20977075220](https://github.com/hertzg/rtl_433_docker/actions/runs/20977075220) | 23m 24s* | 13/13 | `8d92cdd` |
| 2026-01-15 | [21014331812](https://github.com/hertzg/rtl_433_docker/actions/runs/21014331812) | 2m 19s | 13/13 | `8d92cdd` |
| 2026-01-16 | [21050647447](https://github.com/hertzg/rtl_433_docker/actions/runs/21050647447) | 2m 7s | 13/13 | `8d92cdd` |
| 2026-01-17 | [21084634021](https://github.com/hertzg/rtl_433_docker/actions/runs/21084634021) | 2m 27s | 13/13 | `8d92cdd` |
| 2026-01-18 | [21102892264](https://github.com/hertzg/rtl_433_docker/actions/runs/21102892264) | 2m 10s | 13/13 | `8d92cdd` |

All 6 builds used the same upstream rtl_433 commit [8d92cdd](https://github.com/merbanan/rtl_433/commit/8d92cdd60def611865fb2cf2c4b3c7cdb4faf587) ("Add support for Geevon TX19-1 temperature humidity sensor" from 2026-01-06) since there were no new commits to that repo.

All builds show 13/13 Docker layers cached and complete in ~2 minutes. The Jan 14 outlier (23m) was due to slow arm/v7 emulated builds, not cache misses.

The caching fix is working - when upstream hasn't changed, layers are fully cached and no new images are pushed.

*slow arm/v7 emulation on that run
